app/src/main/java/org/simple/clinic/textInputdatepicker/TextInputDatePickerValidator.kt

```diff
@@ -0,0 +1,49 @@
+package org.simple.clinic.textInputdatepicker
+
+import androidx.annotation.VisibleForTesting
+import org.simple.clinic.util.UserClock
+import org.simple.clinic.util.toUtcInstant
+import org.simple.clinic.textInputdatepicker.TextInputDatePickerValidator.Result.Notvalid.DateIsInPast
+import org.simple.clinic.textInputdatepicker.TextInputDatePickerValidator.Result.Notvalid.InvalidPattern
+import org.simple.clinic.textInputdatepicker.TextInputDatePickerValidator.Result.Notvalid.MaximumAllowedDateRange
+import java.time.Instant
+import java.time.LocalDate
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import java.time.format.DateTimeParseException
+import javax.inject.Inject
+import javax.inject.Named
+
+class TextInputDatePickerValidator @Inject constructor(
+    private val userClock: UserClock,
+    @Named("exact_date") private val dateOfBirthFormat: DateTimeFormatter
```

Comment:
Can we rename this variable to something else? We are not validating the date of birth here.